packages/admin/cms-admin/src/contentScope/Provider.tsx

```diff
@@ -126,25 +134,31 @@ export interface ContentScopeProviderProps<S extends ContentScopeInterface = Con
     defaultValue: S;
     values: ContentScopeValues<S>;
     children: (p: { match: match<NonNullRecord<S>> }) => ReactNode;
-    location?: {
-        createPath: (scope: ContentScopeValues<S>) => string;
-        createUrl: (scope: S) => string;
-    };
+    location?: ContentScopeLocation<S>;
 }
 
 export function ContentScopeProvider<S extends ContentScopeInterface = ContentScopeInterface>({
     children,
     defaultValue,
     values,
-    location = { createPath: defaultCreatePath, createUrl: defaultCreateUrl },
+    location = defaultContentScopeLocation,
 }: ContentScopeProviderProps<S>) {
     const path = location.createPath(values);
     const defaultUrl = location.createUrl(defaultValue);
     const match = useRouteMatch<NonNullRecord<S>>(path);
     const [redirectPathAfterChange, setRedirectPathAfterChange] = useState<undefined | string>("");
 
     return (
-        <Context.Provider value={{ path, redirectPathAfterChange, setRedirectPathAfterChange, values }}>
+        <Context.Provider
+            value={{
+                path,
+                redirectPathAfterChange,
+                setRedirectPathAfterChange,
+                values,
+                // @ts-expect-error type mismatch because React Context can't be generic and default value is less specific
```

Comment:
can't you change the type using `as` or so?

Reply:
Of course: e9abb3978f8f8da23c2bd1b66419f486b26ec53f. Is this better than ts-expect-error?

Reply:
_everything_ is better than ts-expect-error :)